source/solvers/heat_transfer_assemblers.cc

```diff
@@ -471,6 +484,16 @@ HeatTransferAssemblerRobinBC<dim>::assemble_rhs(
                       const double T_face =
                         scratch_data.temperature_face_value[f][q];
                       const double JxW = scratch_data.face_JxW[f][q];
+                      const double h =
+                        h_function.value(scratch_data.quadrature_points[q]);
+                      const double T_inf =
+                        T_inf_function.value(scratch_data.quadrature_points[q]);
+                      const double emissivity = emissivity_function.value(
+                        scratch_data.quadrature_points[q]);
+                      Assert(emissivity <= 1.0 && emissivity >= 0.0,
+                             EmissivityError(emissivity));
```

Comment:
Since  you are declaring an exception don't you want to call AssertThrow instead of just Assert?